### DISCUSSION TITLE
Missing Meetings In Control Panel Sorting By Region

### DISCUSSION THREAD

**Discussion by COMPUMEC** [code4recovery/12-step-meeting-list#871] [Q&A] [answered]
When sorting meetings by region in the control panel for "Portland" or "District 1" and so on, it displays only a limited number of  meetings, while on the website you can see many more. 

What is being displayed in the control panel is inaccurate. 

Why is this happening?

Watch video
https://www.dropbox.com/s/mwggvov7mw5e6m1/Dashboard%20%E2%80%B9%20Central%20Service%20Office%20%E2%80%94%20WordPress%20-%20Google%20Chrome%202022-07-12%2018-45-03.mp4?dl=0

**Comment by brianw-area24** [comment]
I am not able to recreate the problem on my test site. I imported your meetings (880 of them), and filtered for Portland, ME, and I got 196, where in your video it said your site found 26 filtering for Portland, ME.

However, I noticed a couple other things in your video:
- Although it says 26 found in the filter, it appears to have found more than that. When you switched to page 2, it showed more than 6 meetings, page 1 was 20 meetings, so if it's only finding 26 meetings, page 2 should have shown 6 meetings, however, it showed 20 meetings also.
- When you first went to the meetings page, it says "2 items" and doesn't show any paging.  It should say 880 items and lots of pages.

Because that number of items is incorrect, even before using the Filter option, I don't think it's a filtering problem. I'm wondering if one of the other plugins on your site might be causing a problem? You might try disabling the other plugins to see if the meetings page starts working correctly.

**Comment by COMPUMEC** [comment]
I did as you suggested with no change. 
I made a new video with more info that may help. 
Not sure if there is a conflict in Districts and Regions?

https://www.dropbox.com/s/xa7q8zi8332nmtf/Dashboard%20%E2%80%B9%20Central%20Service%20Office%20%E2%80%94%20WordPress%20-%20Google%20Chrome%202022-07-15%2015-26-42.mp4?dl=0

**Reply by brianw-area24** [reply]
I'm curious. If you have a different browser on your computer, do you get the same results using a different browser?

**Comment by COMPUMEC** [comment]
I spent the weekend cleaning up my .csv file.
I moved Regions and Sub-Regions to District and Sub-District.
I also went through and cleaned up all meeting notes.

Now when I import my clean list, deleting all previous data, it does not import properly.
The districts and sub-district don't come through. It sorts everything by town in regions with no parent.

If I export my clean csv file after importing it, it exports and moves data.
It moves some of the meeting types to meeting notes.
It messes up the district sub-district.

Please can you figure this out?

[Exported and wrong-meetings.csv](https://github.com/code4recovery/12-step-meeting-list/files/9135346/Exported.and.wrong-meetings.csv)
[Clean-meetings.csv](https://github.com/code4recovery/12-step-meeting-list/files/9135347/Clean-meetings.csv)

**Comment by kiyote33** [comment]
Hi @COMPUMEC Bob,
I'll take a stab at  this. I've taken a look at your production [Area 28 Meeting List](https://csoaamaine.org/meetings/), which appears to have the listing set with the cities/towns located in your area assigned to district regions, in a child/parent relationship. I've made an assumption that this is correct, and that you would like your CSV file configured to repeat this setup in a test/staging environment. Importing into my sandbox, the configuration of your Clean-meetings.csv file created each detected region as a top-level region with no child/parent relationship(see note below). To create the desired parent child relationship, I moved the data beneath the District/Sub District columns to the Region/Sub Region columns respectively. When this modified csv was imported into my sandbox (after deleting everything), it produced the expected results. I did notice that some of the Regions did not align with those on the Area 28 production site. The sandbox results appear consistent with what was assigned in your Clean-meetings.csv. For instance, the region South Portland has been assigned to Districts 24/26 like on the production site, but because none have been assigned to District 01 the child relationship to that district was not created.
Note: For the District/Sub District import logic to work correctly, you would need to have each meeting assigned to a Group. As that field was left blank in the csv, the only other option was to build the hierarchical relationships using the Region/Sub Region field.
Hope this helps. Here's a screenshot of my sandbox: 
![image](https://user-images.githubusercontent.com/57125427/179845132-af13e28a-4694-4425-9f47-ff42ef8670b4.png)

**Reply by COMPUMEC** [reply]
Thank you for all your work on this Wayne!
I still cant get it to go.
I am going to keep working on it and will report back with the results.

**Comment by COMPUMEC** [comment, marked as answer]
kiyote33 
This worked as you described. 
I did get it to work. 
But I simplified it with just 1 region which is the towns.
Less is sometimes better.
